### PR TITLE
Move to sklearn gmm implementation

### PR DESCRIPTION
**Pull request by aarmey**
@mcreixell—I'm interested in what you think of this. Besides this "using" the missing values (which I have a solution for), do you see/foresee any issues? I tried this because I'm concerned about potential issues within pom. This approach isn't giving me any failures up to much higher numbers of clusters, and it's possible to make much faster for large numbers of clusters by calculating all the clusters at once. (Pom loops over peptides and clusters in each iteration.)

### CODE REVIEW

**Comment by mcreixell** [comment]
This would make our method so much more stable! I remember thinking about this a while back because I noticed that with sklearn's GMM was faster, more stable, and I could fit much larger number of clusters but we decided to stick to pom because of the missingness. I'm definitely interested in this change.

**Comment by mcreixell** [comment]
So the idea is filling the missing values with PCA imputation and then using sklearn's GMM?

**Comment by mcreixell** [comment]
How would this affect the manuscript?

**Comment by aarmey** [comment]
Ok great. I'll try and get the rest hooked up when I have a chance.

Actually, I plan to add the GMM imputing missing values directly, but it will use PCA-based imputation as a very good starting point. There's also a [super high-performance version](https://github.com/meyer-lab/tensorPack/blob/main/tensorpac/soft_impute.py) of PCA imputation we can use.

**Comment by aarmey** [comment]
@mcreixell one thing I could use from you on this branch is all the pickle loads removed, so that it just calls the model from scratch. I tried removing one in FigureS4, and it runs, but I'm not sure about the import settings you used in each place. The CPTAC data also seems to be imported differently.

**Comment by aarmey** [comment]
(Not urgent—just whenever.)

**Comment by aarmey** [comment]
> How would this affect the manuscript?

It'd be great to swap this out in the revision, but I don't want to wait on resubmitting for this. I'll keep this separate. If we resubmit without this, we can move it to a separate repo.

**Comment by aarmey** [comment]
> So the idea is filling the missing values with PCA imputation and then using sklearn's GMM?

Yep.

**Comment by mcreixell** [comment]
> @mcreixell one thing I could use from you on this branch is all the pickle loads removed, so that it just calls the model from scratch. I tried removing one in FigureS4, and it runs, but I'm not sure about the import settings you used in each place. The CPTAC data also seems to be imported differently.

Ok, will do. 

> It'd be great to swap this out in the revision, but I don't want to wait on resubmitting for this. I'll keep this separate. If we resubmit without this, we can move it to a separate repo.

I'm definitely interested in resubmitting as soon as possible but also see the benefit in making this change in the revision. It could probably help with some of the "harder" comments to address. We can decide once we get this completely hooked up.

**Comment by mcreixell** [comment]
I added new fittings in every figure file, didn't delete the pickle files yet tho. However, I think the reason why we were pickling models still remains an issue here. Even with sklearn's GMM the clusters will be different in every run.

**Comment by aarmey** [comment]
> I added new fittings in every figure file, didn't delete the pickle files yet tho. However, I think the reason why we were pickling models still remains an issue here. Even with sklearn's GMM the clusters will be different in every run.

I'm confident I can get it to match between runs. :-)

**Comment by mcreixell** [comment]
Fixed why the PCA function in figureM3 was failing and got figures M3A-D to look pretty much identical to what we had (AKT prediction). I added all figures back to the `makefile` to see if they don't fail (except figureM4 which will take many hours to build and will work on this on a different branch). The build of figureM2 will tell me what number of clusters & weight to use for the rest of the figures. I will add a "final" figureM2 here and work on the rest of the figures after merging this.

File: msresist/clustering.py [hunk 1]
```diff
@@ -19,12 +19,13 @@ class MassSpecClustering(GaussianMixture):
...
     should have a larger effect on the peptide assignment. """
-    def __init__(self, info, ncl, SeqWeight, distance_method):
-        super().__init__(n_components=ncl, covariance_type="diag", n_init=2, max_iter=200, tol=1e-4)
+    def __init__(self, info, ncl, SeqWeight, distance_method, random_state=None):
+        super().__init__(n_components=ncl, covariance_type="diag", n_init=2, max_iter=200, tol=1e-4, random_state=random_state)
 
         self.info = info
         self.SeqWeight = SeqWeight
         self.distance_method = distance_method
+        self.random_state = random_state
```
Comment (mcreixell) [inline msresist/clustering.py:28]: Setting the argument `random_state` to an integer makes DDMC to be reproducible (whereas `None` makes the seeding random) so I added this argument. I figured that could be useful to easily switch. @aarmey

Comment (aarmey) [reply msresist/clustering.py:28]: Excellent—yes!

**Comment by mcreixell** [comment]
Hi Aaron, did you cancel the build?

**Comment by aarmey** [comment]
No, looks like it hit the 6 hour limit.

**Comment by aarmey** [comment]
Also looks like output/biol/figureM5.svg failed.

**Comment by mcreixell** [comment]
It failed because I was trying to plot a peptide from the pickled cluster 11 and the new cluster 11 didn't have it. I I'll just build figureM2 and remove the rest of the figures from the makefile for now. Then work on each one of them on separate branches. The fitting works great it's just a matter of updating the functions looking at specific clusters.

**Comment by aarmey** [comment]
Sounds good.

**Comment by aarmey** [comment]
I stopped M2 from building as it was taking a long time. Anything to wait on here before merging?

**Comment by mcreixell** [comment]
Nothing to wait on here. I can take it from here, thanks.

**Pull request merged**